fiftyone/core/metadata.py

```diff
@@ -382,28 +389,79 @@ def compute_metadata(
             default value ``fiftyone.config.show_progress_bars`` (None), or a
             progress callback function to invoke instead
     """
-    num_workers = fou.recommend_thread_pool_workers(num_workers)
+    logger.info("Computing metadata...")
 
     if sample_collection.media_type == fom.GROUP:
         sample_collection = sample_collection.select_group_slices(
             _allow_mixed=True
         )
 
-    if num_workers <= 1:
-        _compute_metadata(
-            sample_collection, overwrite=overwrite, progress=progress
-        )
-    else:
-        _compute_metadata_multi(
+    if not overwrite:
+        sample_collection = sample_collection.exists("metadata", False)
+
+    if (sample_count := len(sample_collection)) == 0:
```

Comment:
Don't compute `len(sample_collection)`! This may cause a full collection scan on views which can take significant time for complex ones on big datasets.